static/app/components/core/compactSelect/listBox/index.tsx

```diff
@@ -163,50 +170,140 @@ export function ListBox({
+      <div
+        ref={virtualizer.scrollElementRef}
+        style={{
+          height: '100%',
+          overflowY: 'auto',
+        }}
       >
-        {overlayIsOpen &&
-          listItems.map(item => {
-            if (item.type === 'section') {
-              return (
-                <ListBoxSection
-                  key={item.key}
-                  item={item}
-                  listState={listState}
-                  hiddenOptions={hiddenOptions}
-                  onToggle={onSectionToggle}
-                  size={size}
-                  showSectionHeaders={showSectionHeaders}
-                  showDetails={showDetails}
-                />
-              );
-            }
-
-            return (
-              <ListBoxOption
-                key={item.key}
-                item={item}
-                listState={listState}
-                size={size}
-                showDetails={showDetails}
-              />
-            );
-          })}
-
-        {!hasSearch && hiddenOptions.size > 0 && (
-          <SizeLimitMessage>
-            {sizeLimitMessage ?? t('Use search to find more options…')}
-          </SizeLimitMessage>
-        )}
-      </ListWrap>
+        <div {...virtualizer.wrapperProps}>
+          <ListWrap
+            {...mergedProps}
+            style={{
+              ...mergedProps.style,
+              ...virtualizer.listWrapStyle,
+            }}
+            onKeyDown={onKeyDown}
+            onMouseLeave={onMouseLeave}
+            ref={mergeRefs(listElementRef, ref)}
+          >
+            {overlayIsOpen &&
+              virtualizer.items.map(row => {
+                const item = listItems[row.index]!;
+                if (item.type === 'section') {
+                  return (
+                    <ListBoxSection
+                      {...virtualizer.itemProps(row.index)}
+                      key={item.key}
+                      item={item}
+                      listState={listState}
+                      hiddenOptions={hiddenOptions}
+                      onToggle={onSectionToggle}
+                      size={size}
+                      showSectionHeaders={showSectionHeaders}
+                      showDetails={showDetails}
+                    />
+                  );
+                }
+
+                return (
+                  <Fragment key={item.key}>
+                    <ListBoxOption
+                      key={item.key}
+                      {...virtualizer.itemProps(row.index)}
+                      item={item}
+                      listState={listState}
+                      size={size}
+                      showDetails={showDetails}
+                    />
+                  </Fragment>
+                );
+              })}
+
+            {!hasSearch && hiddenOptions.size > 0 && (
+              <SizeLimitMessage>
+                {sizeLimitMessage ?? t('Use search to find more options…')}
+              </SizeLimitMessage>
```

Comment:
A discussion for another time, but I think this doesn't do a good job at explaining that that the input is async

Reply:
I don't see a reason why you'd want to do that now that virtualization can render all items? It feels like an artifact of the past that we could remove in a followup, wdyt?